distribution/lib/Standard/Base/0.0.0-dev/src/Runtime.enso

```diff
@@ -58,13 +60,18 @@ gc = @Builtin_Method "Runtime.gc"
    Asserts that the given action succeeds, otherwise throws a panic.
 
    Assertions are disable by default, meaning that call to this method is
-   a no-op. To enable assertions, set the environment variable
-   `ENSO_ENABLE_ASSERTIONS=true`
-assert : Boolean -> Text -> Nothing ! Assertion_Error
-assert ~action message="" = assert_builtin action message
+   a no-op. To enable assertions, either set the environment variable
+   `ENSO_ENABLE_ASSERTIONS=true` or enable JVM assertions by passing `-ea`
+   cmd line option to java.
+assert : Boolean -> Text -> Nothing
+assert (~action : Boolean) (message : Text = "") =
+    if assertions_enabled.not then Nothing else
```

Comment:
Also note that `assertions_enabled` results in a compilation constant.

Reply:
Should we run some benchmark comparing the old and new implementations?

It may be just intuition, but I was trusting the Truffle Node implementation more to be quickly removed from running, here I'm less convinced.

Reply:
> Should we run some benchmark comparing the old and new implementations?

I am not convinced that is necessary.

> but I was trusting the Truffle Node implementation more to be quickly removed from running, here I'm less convinced

It is the opposite for me. I am convinced that `assertions_enabled` is a compilation constant.

Reply:
I have ensured that `Runtime.assert` is a nop in production after PE. See [my comment on the PR](https://github.com/enso-org/enso/pull/8742#issuecomment-1889112453).